api/v0/proto/retrievability/signer_service.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.signer;
+
+// TODO: this whole service needs to be improved
+
+message SubmitOptions {
+  // Amount to be paid per gas unit.
+  double gas_price = 1;
+  // Gas amount.
+  optional uint64 gas = 2;
+  // Keystore key for signing.
+  string key_name = 3;
+  // Account paying for transaction.
+  optional string fee_granter = 4;
+  // Transaction nonce.
+  optional uint64 nonce = 5;
+}
+
+message SubmitDataRequest {
+  bytes namespace = 1;
+  repeated bytes data = 2;
+  SubmitOptions options = 3;
+}
+
+message SubmitDataResponse {}
+
+// SignerService - handles submitting
+service SignerService {
+  rpc SubmitData(SubmitDataRequest) returns (SubmitDataResponse);
```

Comment:
Why not `SubmitBlob` since most other documentation refers to this as a blob

Reply:
The change from `SubmitBlob` to `SubmitData` does not belong in this PR. It was introduced in the initial version ([v0](https://github.com/celestiaorg/canonical-api/pull/1)). The main rationale was to allow users to submit raw data as `[]byte`, rather than requiring them to construct a `Blob` structure themselves.

Reply:
Blob could still refer to the bytes being submitted not specifically the `Blob` struct after all it literally stands for binary large object

Reply:
It does not take Blob message as input, which is also defined in API.  So I tried to come up with better name to not confuse users. Like it would be theta `GetBlob` returns `Blob`, while SubmitBlob would submit bytes. 

What you think about if we call it just `Submit`?  There is no other Submit in near future

Reply:
I don't think there's any reason to ever have a user submit a blob as shares instead of just the bytes, since it's easy to go from shares to bytes but not the other way around.

Reply:
Fixed in 2c0f016e45a7445551ed8589b3189b8ea7c86d8b